pkgs/by-name/fl/floorp-bin-unwrapped/package.nix

```
@@ -0,0 +1,133 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  wrapGAppsHook3,
+  autoPatchelfHook,
+  alsa-lib,
+  curl,
+  dbus-glib,
+  gtk3,
+  libXtst,
+  libva,
+  pciutils,
+  pipewire,
+  adwaita-icon-theme,
+  patchelfUnstable, # have to use patchelfUnstable to support --no-clobber-old-sections
+  undmg,
+  writeText,
+}:
+
+let
+  inherit (lib.importJSON ./sources.json) version sources;
+
+  binaryName = "floorp";
+  policies = {
+    DisableAppUpdate = true;
+  };
+
+  policiesJson = writeText "floorp-policies.json" (builtins.toJSON { inherit policies; });
+
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "floorp-bin-unwrapped";
+  inherit version;
+
+  src = fetchurl {
+    inherit (sources.${stdenv.hostPlatform.system}) url sha256;
+  };
+
+  sourceRoot = lib.optional stdenv.hostPlatform.isDarwin ".";
+
+  nativeBuildInputs = [
+    wrapGAppsHook3
+  ]
+  ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [
+    autoPatchelfHook
+    patchelfUnstable
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    undmg
+  ];
+
+  buildInputs = lib.optionals (!stdenv.hostPlatform.isDarwin) [
+    gtk3
+    adwaita-icon-theme
+    alsa-lib
+    dbus-glib
+    libXtst
+  ];
+
+  runtimeDependencies = [
+    curl
+    pciutils
+  ]
+  ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [
+    libva.out
+  ];
+
+  appendRunpaths = lib.optionals (!stdenv.hostPlatform.isDarwin) [
+    "${pipewire}/lib"
+  ];
+
+  # Firefox uses "relrhack" to manually process relocations from a fixed offset
+  patchelfFlags = [ "--no-clobber-old-sections" ];
+
+  # don't break code signing
+  dontFixup = stdenv.hostPlatform.isDarwin;
+
+  installPhase = ''
+    runHook preInstall
+  ''
+  + (
+    if stdenv.hostPlatform.isDarwin then
+      ''
+        mkdir -p $out/Applications
+        mv Firefox*.app "$out/Applications/Floorp.app"
```

```suggestion
        mv Floorp.app "$out/Applications/Floorp.app"
```

Fix for
```
building '/nix/store/p5bznpimv1hwiqkl3pk0v3jizglw3zry-floorp-bin-unwrapped-12.1.4.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/9v2f51iy2kafh0ihbdxm4m2rgv874ay2-floorp-macOS-universal.dmg
source root is .
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
no Makefile or custom buildPhase, doing nothing
Running phase: glibPreInstallPhase
Running phase: glibPreInstallPhase
Running phase: installPhase
mv: missing destination file operand after '/nix/store/1d7q14gbwjwsrzdba98l5ad1flinc39a-floorp-bin-unwrapped-12.1.4/Applications/Floorp.app'
Try 'mv --help' for more information.
```

The built package doesn't start though